airtime_mvc/application/controllers/BillingController.php

```diff
@@ -19,6 +20,24 @@ public function indexAction()
         $this->_redirect('billing/upgrade');
     }
 
+    public function promoEligibilityCheckAction()
+    {
+        $this->view->layout()->disableLayout();
+        $this->_helper->viewRenderer->setNoRender(true);
+
+        $request = $this->getRequest();
+        if (!$request->isPost()) {
+            throw new Exception("Must POST data to promoEligibilityCheckAction.");
+        }
```

Comment:
On the PHP side, in the billing form, add:

$csrf_namespace = new Zend_Session_Namespace('csrf_namespace');
$csrf_element = new Zend_Form_Element_Hidden('csrf');
$csrf_element->setValue($csrf_namespace->authtoken)->setRequired('true')->removeDecorator('HtmlTag')->removeDecorator('Label');
$form->addElement($csrf_element);

In the template, you can put:
$this->form->getElement('csrf')

In Javascript, you can add this to the dictionary: 
    "csrf_token" : $("#csrf").attr('value')